rclcpp_components/include/rclcpp_components/component_manager.hpp

```diff
@@ -233,6 +233,8 @@ class ComponentManager : public rclcpp::Node
 
 private:
   std::weak_ptr<rclcpp::Executor> executor_;
+  std::map<uint64_t, std::shared_ptr<rclcpp::Executor>> dedicated_executors_;
+  std::map<uint64_t, std::thread> dedicated_threads_;
```

Comment:
do the maps need to be ordered?
It doesn't look like they need, use `std::unordered_map` in that case